src/videojs-hls.js

```diff
@@ -906,9 +915,39 @@ videojs.HlsHandler.prototype.fillBuffer = function(mediaIndex) {
     // the state of the buffer before a segment is appended will be
     // stored here so that the actual segment duration can be
     // determined after it has been appended
-    buffered: null
+    buffered: null,
+    // The target timestampOffset for this segment when we append it
+    // to the source buffer
+    timestampOffset: null
   };
 
+  if (mediaIndex > 0) {
+    segmentTimestampOffset = videojs.Hls.Playlist.duration(segmentInfo.playlist,
+      segmentInfo.playlist.mediaSequence + mediaIndex) + this.playlists.expired_;
+  }
+
+  // If we have seeked into a non-buffered time-range, remove all buffered
+  // time-ranges because they could have been incorrectly placed originally
```

Comment:
This comment seems out of date (specifically, the bits about clearing the buffer aren't at least out of order in this new location)